src/findReplace.js

```diff
@@ -219,147 +220,39 @@ export default class FindReplace extends Plugin {
-						}
-					}
-				}
-				// update markers variable after search
-				markers = Array.from( model.markers.getMarkersGroup( SEARCH_MARKER ) );
-			} );
-			this.currentSearchIndex = 0;
-		}
-		const currentMarker = markers[ this.currentSearchIndex ];
-		this._scrollTo( currentMarker );
+		const findText = findField.fieldView.element.value;
+		const {currentMarker,markers,currentIndex,total} = this.editor.execute('findReplace',{
+			findText,
+			increment
+		})
 		const t = this.editor.t;
 		if ( markers && markers.length && markers.length > 0 ) {
-			findField.infoText = this.currentSearchIndex + 1 + t( ' of ' ) + markers.length;
+			findField.infoText = currentIndex + 1 + t( ' of ' ) + total;
 		} else {
 			findField.infoText = t( 'Not found' );
 		}
 		return currentMarker;
 	}
 
-	_scrollTo( marker ) {
-		const editor = this.editor;
-		if ( marker ) {
-			editor.model.change( writer => {
-				this._removeCurrentSearchMarker( writer );
-				this.currentSearchMarker = writer.addMarker( CURRENT_SEARCH_MARKER,
-					{ range: marker.getRange(), usingOperation: false } );
-			} );
-			const viewRange = editor.editing.mapper.toViewRange( marker.getRange() );
-			const domRange = editor.editing.view.domConverter.viewRangeToDom( viewRange );
-			scrollViewportToShowTarget( { target: domRange, viewportOffset: 130 } );
-		}
-	}
-
 	_resetStatus() {
 		this.findInput.infoText = undefined;
 		this.replaceInput.infoText = undefined;
-		this.currentSearchIndex = 0;
-		const model = this.editor.model;
-		model.change( writer => {
-			for ( const searchMarker of model.markers.getMarkersGroup( SEARCH_MARKER ) ) {
-				writer.removeMarker( searchMarker );
-			}
-			this._removeCurrentSearchMarker( writer );
-		} );
+		removeSearchMarkers( this.editor.model );
 	}
 
-	_removeCurrentSearchMarker( writer ) {
-		if ( this.currentSearchMarker ) {
-			writer.removeMarker( this.currentSearchMarker );
-			this.currentSearchMarker = undefined;
-		}
-	}
-
-	_replace( findField, replaceField ) {
-		const model = this.editor.model;
-		const markers = Array.from( model.markers.getMarkersGroup( SEARCH_MARKER ) );
-		const sameSearch = this._isSameSearch( findField, markers );
-		const currentMarker = sameSearch ? markers[ this.currentSearchIndex ] : this._find( findField, 1 );
-		const replaceBy = replaceField.fieldView.element.value;
-		if ( currentMarker && currentMarker.getRange ) {
-			model.change( writer => {
-				model.insertContent( writer.createText( replaceBy ), currentMarker.getRange() );
-				writer.removeMarker( currentMarker );
-				this._removeCurrentSearchMarker( writer );
-			} );
-			// refresh the items...
-			this._find( findField, 0 );
-		}
+	_replace( findField, replaceField,replaceAll=false ) {
+		const findText = findField.fieldView.element.value;
+		const replaceText = replaceField.fieldView.element.value;
+		this.editor.execute('findReplace',{
+			findText,
+			replaceText,
+			replaceAll
+		})
+		this._find(findField,0)
```

Comment:
@YueHui do we really need to call `this._find` here? It seems it is already called inside FindCommand. Could you please check if it is really necessary?

Reply:
the purpose of call `this._find` is to update the `findField` infotext,it's Inappropriate, i fixed it